lib/coverband/reporters/web.rb

```diff
@@ -9,71 +9,74 @@ class Web
       attr_reader :request
 
       def initialize
+        full_path = Gem::Specification.find_by_name('simplecov-html').full_gem_path
         @static = Rack::Static.new(self,
-          root: File.expand_path('public', Gem::Specification.find_by_name('simplecov-html').full_gem_path),
-          urls: [/.*\.css/, /.*\.js/, /.*\.gif/, /.*\.png/]
-        )
+                                   root: File.expand_path('public', full_path),
+                                   urls: [/.*\.css/, /.*\.js/, /.*\.gif/, /.*\.png/])
       end
 
       def call(env)
         @request = Rack::Request.new(env)
 
         if request.post?
           case request.path_info
-          when /\/collect_update_and_view/
+          when %r{\/collect_update_and_view}
             collect_update_and_view
-          when /\/clear/
+          when %r{\/clear}
```

Comment:
Not s show stopper but it doesn't feel right that everyone on that team that can view the coverage report can also clear all the data. Right now it's difficult to know how much to believe in the coverage data because we don't know how long it's been collecting the information.

Soon I think we should merge in the change that uses a file's checksum to automatically reset coverage data on a file by file basis. Then i think we can remove the idea of clearing the data periodically from redis.

Reply:
yeah good call. I understand the confusion around how long / valid is the data.

1. I think we can add an option to disable this feature, then the rake task would have to be used to clear. I think that can land on the 3.0.x line
2. Agreed on getting to a checksum or git release tag type solution. My biggest concern is around the performance of calculating all the hashes on the fly. Might be able to get in 3.0.x or might make that one of the major features to drive 4.0

Reply:
RE: Performance of checksum. I did some really basic benchmarking calculating user.rb file checksum 1000X and it's very fast:

```
irb(main):006:0> puts Benchmark.measure { 1000.times { Digest::MD5.file('./app/models/user.rb').hexdigest } }
  0.020000   0.010000   0.030000 (  0.027000)
```
That's 0.000027  seconds per md5. In addition, you probably only need to calculate the md5 one time in production as source file md5's do not change after the process starts. That makes it even faster:

```
irb(main):007:0> checksum_cache = {}
=> {}
irb(main):008:0> puts Benchmark.measure { 1000.times { checksum_cache['./app/models/user.rb'] ||= Digest::MD5.file('./app/models/user.rb').hexdigest } }
  0.000000   0.000000   0.000000 (  0.001463)
=> nil
irb(main):009:0>
```
Which is 0.000001463 seconds per md5.

At those numbers, i wonder if it makes sense to fold this feature into 3.0.x? It really takes a major complication away.

Reply:
OK I will move that back to the 3.0.x so long as we don't have any API breaking changes. Thinking still after 3.0.0.